src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1435,3 +1435,45 @@ function CategoricalArrays.categorical(df::AbstractDataFrame,
     end
     DataFrame(newcols, _names(df), copycols=false)
 end
+
+"""
+    flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
+
+When column `veccol` of `df` has iterable elements that define `length`, for example a `Vector` 
+of `Vector`s. Returns a DataFrame where each element of `veccol` is flattened. 
+Elements of row `i` of `df` other than `veccol` will be duplicated according to 
+the length of `df[i, veccol]`
+
+# Arguments
```

Comment:
No need for this section as it repeats what has been said above. But an example would be useful.